nsqd/nsqd.go

```diff
@@ -370,6 +379,11 @@ func (n *NSQD) PersistMetadata() error {
 		return err
 	}
 
+	err = os.Remove(oldFileName)
```

Comment:
My preference is (a), I dislike the idea of maintaining compatibility (and this symlink proposal sounds questionable).

That said, I don't think it's necessary to remove the the old file? If we leave it alone, the rollback path becomes easier.